Fix Gitpod config (Python + Vale) and Vale vocab handling

dennisameling opened the pull request:
This fixes many Gitpod-related issues and allows for a relatively smooth user experience again. Ideally, in the future I'd like us to have a custom Dockerfile that installs these dependencies instead. While this is possible today in combination with Gitpod, it can't cache Dockerfile builds (yet?) resulting in slow boot times. The current approach of installing the dependencies after the basic `python:3.9` image is loaded is the fastest at the moment.

Fixes #16

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/21"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

dennisameling commented:
@RCheesley this fixes a bunch of issues with using Gitpod. I also noticed that Mautic's Vale vocab wasn't picked up correctly - I just fixed that by properly renaming/moving the existing vocab file according to [Vale's docs](https://docs.errata.ai/vale/vocab) 👍🏼 

Note that the Vale GitHub Action is still configured to only check "changed files" instead of the entire documentation - I've created https://github.com/mautic/developer-documentation-new/issues/22 so we can enable it for the entire documentation soon! 🚀

RCheesley commented:
When I test this in Gitpod I still get the error that vale is not known - and it asks me to install Python and Vale in popups. Is that expected behaviour @dennisameling ?

dennisameling commented:
@RCheesley regarding the Python extension question:
- We install VS Code extension in the Gitpod configuration - https://github.com/mautic/developer-documentation-new/blob/76280e2ea7131ed41519548992364c002b768f30/.gitpod.yml#L14-L18
- Then there's also https://github.com/mautic/developer-documentation-new/blob/main/.vscode/extensions.json where we "recommend" extensions in case folks checkout the repo locally. I've just removed the recommendations for now and will create an issue with the Gitpod team for a smoother user experience here.

Then, regarding the Vale message: the problem is that Vale only gets installed after the VS Code workspace has been opened already, hence the message. I've just switched to using a custom Dockerfile which has Python + Vale preinstalled, so that both are available on workspace boot.

Could you please try again now? 🙏🏼
